src/map/storage.cpp

```diff
@@ -603,7 +603,12 @@ char storage_guild_storageopen(struct map_session_data* sd)
 		return GSTORAGE_ALREADY_OPEN;
 	}
 
-	if((gstor = guild2storage2(sd->status.guild_id)) == NULL) {
+	if((gstor = guild2storage2(sd->status.guild_id)) == nullptr
+#ifdef OFFICIAL_GUILD_STORAGE
+		|| (gstor && gstor->max_amount < guild_checkskill(sd->guild, GD_GUILD_STORAGE) * 100)
```

Comment:
Checking gstor for not null in this "or" condition is useless, since it will short-circuit if the check before returns null. Hence gstor can only be not null after this check.
Also I would not only check if max_amount is smaller, but if its not identical. In case the guild skills got reseted.